vxquery-core/src/main/java/org/apache/vxquery/compiler/rewriter/rules/AbstractCollectionRule.java

```diff
@@ -125,6 +125,91 @@ protected String getCollectionName(Mutable<ILogicalOperator> opRef) throws Algeb
         return collectionName;
     }
 
+    protected String[] getCollectionWithTagName(Mutable<ILogicalOperator> opRef) throws AlgebricksException {
```

Comment:
Can we share code with getCollectionName? If getCollectionName returned a list of arguments this could help with sharing code. Then you would only need to specify the specific collection function you are using.
SideNote: I think the OperatorToolbox and ExpressionToolbox could simplify this code here.